source/common/buffer/watermark_buffer.h

```diff
@@ -54,6 +54,7 @@ class WatermarkBuffer : public LibEventInstance {
   void postProcess() override { checkLowWatermark(); }
 
   void setWatermarks(uint32_t low_watermark, uint32_t high_watermark);
+  uint32_t high_watermark() const { return high_watermark_; }
```

Comment:
FWIW I have no great opinion on this one way or the other but @dnoe is correct that is what most of the code is doing.